### PR TITLE
Fix test dir not exist when run single test

### PR DESCRIPTION
**Pull request by laura-ding**

### CODE REVIEW

File: tests/common/nebula_service.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class NebulaService(object):
     def __init__(self, build_dir, src_dir):
         self.build_dir = build_dir
         self.src_dir = src_dir
-        self.work_dir = os.path.join(os.getcwd(), 'build', 'server')
+        self.work_dir = os.path.join(os.getcwd(), 'build', 'server_' + time.strftime("%Y-%m-%dT%H:%M:%S", time.localtime()) )
```
Comment (jievince) [inline tests/common/nebula_service.py:25]: Why not use `self.work_dir = os.path.join(build_dir, 'server_.....')`?  If so, `os.mkdir` will work well.

File: tests/common/nebula_service.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def _telnet_port(self, port):
     def install(self):
         if os.path.exists(self.work_dir):
             shutil.rmtree(self.work_dir)
-        os.mkdir(self.work_dir)
+        os.makedirs(self.work_dir)
```
Comment (Shylock-Hg) [inline tests/common/nebula_service.py:88]: Don't do this, it's feature that don't allow to create server if `build` folder don't exists.

File: tests/common/nebula_service.py (hunk 1 above)
Comment (laura-ding) [reply tests/common/nebula_service.py:25]: I understand that was running the test, where generating the dir. Your understanding is to be generated in a fixed place. it ok for them.

File: tests/common/nebula_service.py (hunk 2 above)
Comment (laura-ding) [reply tests/common/nebula_service.py:88]: So your path generation is wrong, you created it with the current directory, not the build directory as the parent node. you can make it as @jievince said.

Comment (Shylock-Hg) [reply tests/common/nebula_service.py:88]: Yes, build_dir better.

**Review by jievince — APPROVED** [review]

**Review by dutor — APPROVED** [review]

**Pull request merged**